pkg/keyspace/tso_keyspace_group.go

```diff
@@ -618,15 +622,7 @@ func buildSplitKeyspaces(
 				oldSplit = append(oldSplit, keyspace)
 			}
 		}
-		// Dedup new keyspaces if it's necessary.
```

Comment:
we don't need this check?

Reply:
I found that `newNum` is always equal to `len(newKeyspaceMap)`,  so we can directly return `new` as the result here.

Reply:
We can't remove this check. I specifically add this block to fix issue https://github.com/tikv/pd/issues/6687. 

root@serverless-cluster-pd-0:/# ./pd-ctl keyspace-group split 0 2 **2 2**
Success!

{"id":2,"user-kind":"basic","members":[{"address":"http://pd-tso-server-0.tso-service.tidb-serverless.svc:2379","priority":200},{"address":"http://pd-tso-server-1.tso-service.tidb-serverless.svc:2379","priority":300}],"keyspaces":**[2,2]**}

Reply:
It's my fault, since I signed off this change and previous change didn't add unittest.

Reply:
I'll reopen #6687.